src/verify.rs

```diff
@@ -5,25 +5,52 @@ use policy_fetcher::verify::Verifier;
 use std::io::prelude::*;
 use std::{collections::HashMap, fs::File};
 
+pub(crate) type VerificationAnnotations = HashMap<String, String>;
+
 pub(crate) async fn verify(
     url: &str,
-    docker_config: Option<DockerConfig>,
-    sources: Option<Sources>,
-    annotations: Option<HashMap<String, String>>,
+    docker_config: &Option<DockerConfig>,
+    sources: &Option<Sources>,
+    annotations: &Option<VerificationAnnotations>,
     key_file: &str,
+) -> Result<String> {
+    let verification_key = read_key_file(key_file)?;
+    let mut verifier = Verifier::new(sources.clone());
+    let verified_manifest_digest = verifier
+        .verify(
+            url,
+            docker_config.clone(),
+            annotations.clone(),
+            &verification_key,
+        )
+        .await?;
+
+    println!("Policy successfully verified");
+    Ok(verified_manifest_digest)
+}
+
+pub(crate) async fn verify_local_checksum(
+    url: &str,
+    docker_config: &Option<DockerConfig>,
+    sources: &Option<Sources>,
+    verified_manifest_digest: &str,
 ) -> Result<()> {
+    let mut verifier = Verifier::new(sources.clone());
+    verifier
+        .verify_local_file_checksum(url, docker_config.clone(), verified_manifest_digest)
+        .await?;
+
+    println!("Local checksum successfully verified");
+    Ok(())
+}
+
+fn read_key_file(path: &str) -> Result<String> {
     let mut pub_key_file =
-        File::open(key_file).map_err(|e| anyhow!("Cannot read verification key file: {:?}", e))?;
+        File::open(path).map_err(|e| anyhow!("Cannot read verification key file: {:?}", e))?;
     let mut verification_key = String::new();
     pub_key_file
         .read_to_string(&mut verification_key)
         .map_err(|e| anyhow!("Error reading contents of verification key file {:?}", e))?;
```

Comment:
Something from the old days, not your fault. This can be done in a more succinct way:

```rust
let contents = std::fs::read_to_string("/etc/motd")
  .map_err(|e| anyhow!("Something went wrong: {:?}", e)?;
```

Reply:
True, thanks! done in https://github.com/kubewarden/kwctl/pull/84/commits/595796007840a6eb43ffd9bc95a004aebe3e98ed